src/mavlink-router/mainloop.cpp

```diff
@@ -646,3 +731,47 @@ bool Mainloop::_retry_timeout_cb(void *data)
 
     return false;
 }
+
+void Mainloop::_handle_pipe()
+{
+    char cmd[1024];
+    ssize_t num_read = read(pipefd, cmd, sizeof(cmd));
+    if (num_read > 0) {
+        cmd[num_read] = 0;
+        log_debug("Pipe read %ld bytes: %s", num_read, cmd);
+        // Op UDP Name IP Port Eavesdropping
+        std::vector<std::string> a;
+        char *pch = strtok(cmd, " ");
+        while (pch != NULL) {
+            a.push_back(std::string(pch));
+            pch = strtok(NULL, " \n");
+        }
+        // Currently only adding dynamic udp endpoints is supported
+        if (a.size() < 6 || a[0] != "add" || a[1] != "udp") {
+            return;
+        }
+        std::string name = a[2];
+        std::string address = a[3];
+        int port = atoi(a[4].c_str());
+        if (port <= 0) {
+            return;
+        }
+
+        std::string command = (std::string)cmd;
```

Comment:
I think this would be nicer without a c-style cast (I think it might convert implicitly anyway)

Reply:
True. Replaced here and on one other place.